### PR TITLE
px upgrade 2.5.1

### PR DESCRIPTION
**Pull request by trierra**

### CODE REVIEW

File: charts/portworx/v1.0.8/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: portworx
-appVersion: 1.0.8
-version: 1.0.8
+appVersion: 1.0.9
+version: 1.0.9
```
Comment (harsh-px) [inline charts/portworx/v1.0.8/Chart.yaml:3]: looks like this needs fixing. We don't want to update the 1.0.8 chart

File: charts/portworx/v1.0.9/questions.yml [hunk 2]
```diff
@@ -0,0 +1,924 @@
...
+  group: "Advanced parameters"
+
+- variable: envVars
+  label: "Environment variables"
+  description: "Enter your environment variables separated by semicolons (example: API_SERVER=http://lighthouse-new.portworx.com;MYENV1=val1;MYENV2=val2). These environment variables will be exported to Portworx."
+  type: string
+  group: "Advanced parameters"
+
+- variable: imageVersion
+  default: "2.5.1"
```
Comment (harsh-px) [inline charts/portworx/v1.0.9/questions.yml:915]: update the PR title (the PR title says 2.5.0)

File: charts/portworx/v1.0.9/templates/portworx-controller.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      {{- if not (empty .Values.registrySecret) }}
+      imagePullSecrets:
+        - name: {{ .Values.registrySecret }}
+     {{- end }}
+      containers:
+      - command:
+        - kube-controller-manager
+        - --leader-elect=true
+        - --address=0.0.0.0
+        - --controllers=persistentvolume-binder
```
Comment (harsh-px) [inline charts/portworx/v1.0.9/templates/portworx-controller.yaml:99]: let's also start the persistentvolume expander controller as that is the one that will handle PVC resize (`--controllers=persistentvolume-binder, persistentvolume-expander`)

**Comment by trierra** [comment]
done

File: charts/portworx/v1.0.9/templates/portworx-controller.yaml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      {{- if not (empty .Values.registrySecret) }}
+      imagePullSecrets:
+        - name: {{ .Values.registrySecret }}
+     {{- end }}
+      containers:
+      - command:
+        - kube-controller-manager
+        - --leader-elect=true
+        - --address=0.0.0.0
+        - --controllers=persistentvolume-binder, persistentvolume-expander
```
Comment (harsh-px) [inline charts/portworx/v1.0.9/templates/portworx-controller.yaml:99]: remove the extra space after comma

File: charts/portworx/v1.0.9/templates/portworx-ds.yaml [hunk 5]
```diff
@@ -0,0 +1,389 @@
+{{/* Setting defaults if they are omitted. */}}
```
Comment (harsh-px) [inline charts/portworx/v1.0.9/templates/portworx-ds.yaml:1]: looks like rancher is still missing the portworx-api ds and service that you recently added to the portworx/helm chart. Let's add that here. It's needed for certain stork features like DR.

**Comment by harsh-px** [comment]
Once above 2 comments are resolved, let's have this run through our internal rancher test job on this branch.

**Comment by trierra** [comment]
done, all tests are passed. @guangbochen  could you please merge this PR?

File: charts/portworx/v1.0.9/questions.yml [hunk 6]
```diff
@@ -0,0 +1,921 @@
...
+  default: "2.3.1"
+  label: "Stork version"
+  type: string
+  group: "Advanced parameters"
+
+- variable: lighthouseVersion
+  default: "2.0.5"
+  type: string
+  label: "Lighthouse version"
+  group: "Advanced parameters"
```
Comment (harsh-px) [inline charts/portworx/v1.0.9/questions.yml:903]: These are outdated. Stork is 2.4.0 and Lighthouse is 2.0.7

Comment (trierra) [reply charts/portworx/v1.0.9/questions.yml]: done

**Comment by harsh-px** [comment]
@PennyScissors could you help merge this?

**Pull request merged**